service/storage_service.cc

```diff
@@ -1104,9 +1104,11 @@ future<> storage_service::handle_state_removing(inet_address endpoint, std::vect
             // will be removed from _replicating_nodes on the
             // removal_coordinator.
             auto notify_endpoint = ep.value();
-            //FIXME: discarded future.
-            (void)restore_replica_count(endpoint, notify_endpoint).handle_exception([endpoint, notify_endpoint] (auto ep) {
+            // OK to discard future since _async_gate is closed on stop()
+            (void)with_gate(_async_gate, [this, endpoint, notify_endpoint] {
+              return restore_replica_count(endpoint, notify_endpoint).handle_exception([endpoint, notify_endpoint] (auto ep) {
                 slogger.info("Failed to restore_replica_count for node {}, notify_endpoint={} : {}", endpoint, notify_endpoint, ep);
+              });
```

Comment:
Maybe it's not a good idea to wait for rpc from within a gossip handler.

Reply:
As @asias said in https://github.com/scylladb/scylla/issues/10672#issuecomment-1139170799
it may take a very long time (hours), and and as commented above:
```
            // No need to wait for
            // restore_replica_count to complete which can take a long time,
            // since when it completes, this node will send notification to
            // tell the removal_coordinator with IP address notify_endpoint
            // that the restore process is finished on this node. This node
            // will be removed from _replicating_nodes on the
            // removal_coordinator.
```